api/src/repositories/sample-method-repository.ts

```diff
@@ -0,0 +1,147 @@
+import SQL from 'sql-template-strings';
+import { z } from 'zod';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { BaseRepository } from './base-repository';
+import { PostSamplePeriod } from './sample-period-repository';
+
+export interface PostSampleMethod {
```

Comment:
Same idea here as my previous comment, re: `SampleMethod` and `SampleMethodRecord` naming